tool/src/org/antlr/v4/semantics/SymbolChecks.java

```diff
@@ -341,42 +318,130 @@ public void checkForModeConflicts(Grammar g) {
 		}
 	}
 
+	public void checkForUnreachableTokens(Grammar g) {
```

Comment:
can you explain in a comment your strategy for finding unreachable tokens?  I also need to review that strategy to know if it is correct. :)

Reply:
Done :)